### PR TITLE
Small traitor objective rework: Protect & kill cross-pollination

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

This changes up traitor objectives a bit. Now, traitors can get protect objectives, and kill and protect objectives will be "cross-pollinated" between traitors.

What this means is, there's a decent chance for one traitor's protect target to be another's assassinate target. With this knowledge, protect targets will be encouraged to actually interact with and keep up with their target, because there's always a chance another traitor is going to actively try to kill/maroon them.

Whenever rolling a kill or protect objective with a random target, the target will be added to the global kill or protect pool.
Every time a protect objective rolls, if there are any people in the "kill pool", there is a 75% chance the target will be picked from the kill pool instead of being random. The target will then most likely be removed from said pool after being selected (although there is a 15% chance they will remain in the pool after being chosen, so having two people protecting a target that one person is trying to kill, or vice versa, is entirely possible).
The same is also true for kill/maroon objectives, except it will pull from the "protect pool" instead.

Just to clarify, this does not _guarantee_ that your kill objective will be someone else's protect objective, and vice versa. It's entirely probable to get random objectives, and for those random objectives to never be picked as another traitor's target.

Traitor protect objectives will be "protect and ensure they escape alive", rather than just "protect".

Steal, research, and kill AI objectives are unchanged.

Also, it's more likely for a kill objective to be a maroon objective instead (40% instead of 30%), because maroon objectives encourage slightly less round removal.

## Why It's Good For The Game

Traitors with protect objectives will be encouraged to actually interact with and keep up with their target, because there's always a chance another traitor is going to actively try to kill/maroon them. This will likely, of course, result in roleplay, as you'd have to come up with excuses to protect your target, without seeming weird.

In addition, this also will require traitors with kill/maroon objectives to plan more, perhaps roleplay more, in order to succeed in their objective.

This could even result in scenarios where you have to hijack the shuttle, while ensuring someone escapes safely on the shuttle or a pod!

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

</details>

## Changelog
:cl:
add: Traitors can now get protect objectives.
balance: Traitors kill or protect objectives have a chance to be chosen from a pool of other traitor's objectives, which may mean your assassination target is being protected by another traitor, or your protect target is being targeted for assassination by another traitor. This is not guaranteed, though.
tweak: It's slightly more likely for a kill objective to be a maroon objective instead.
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
Hmm mixed feelings, i feel like this will lead to a lot of  scenarios where traitors will decide to act "pro-nt" just to protect their objective, specially more if their protection target just happens to be an IC friend, they will have to waste resources and time to hug-box a random crewmember since they will OOC know that there's a chance another traitor might want to kill them

/tg/ has varies on it's antag system where they have different "factions" in which the antag tc-hud tells you stuff alongside "you work alone/you only aid gorlex marauders/you can choose to aid donk co agents or not/everybody is your agent/etcetera"

But we don't have that, we have the Syndicate that can sometimes choose to aid fellow agents or just ignore them and do their own, without a proper introduction of some sort of "factions"  besides the given liberty to choose to follow their objectives or not that's granted when they're antag this would lead to small mini-incursion 1v1 fights where both antags waste their resources on themselves rather than antagonize the crew

We already have mini-issues antagonistically speaking when on incursion the ex-communicated rushes into sec, claims to be followed by syndies, buys a radio and snitch on the entire incursion, i can already predict a player going to sec, claim their allegiance but that they've been tasked to protect X and promise to not cause issues, maybe even revealing code-words or snitch on the other traitors, sure it's interesting *one* round but when it becomes *the meta* it's get boring and stale for both sides

Even if the antag pulls up their .357 to kill an assailant trying to get the protect target, they still will get arrested, stripped of contraband and be left with having to be like a tider-with-saw to protect their target, i feel like it backfires a lot the moment you show yourself and Security, by space law, will not let them play easily.

You can make a few gimmicks sure, but as an antag and sec player this has a lot of holes that are against what our antagonists are supposed to do on Bee.

I do agree having more objectives is a good idea, but i don't feel too certain about this one, i've seen our antag players a lot and how they usually play out it's gonna end up looking like a weird obsessed minus the killing part(except maybe from someone other agent) but with TC, there should be a reason why the Syndicate randomly wants X person to be protected when they have no allegiance with the Syndicate.


> This could even result in scenarios where you have to hijack the shuttle, while ensuring someone escapes safely on the shuttle or a pod!

We're already having too many issues with antagonists hijacking4noraisin and we've been discussing it on another PR and on Discord,  this would just add more paperwork, besides that would imply protecting a random NT crewmember warrants an attempt to make a TDM in shuttle cockpit ?? Doesn't make sense.


As for having a little more marooning objectives, that's okay it opens opportunities of RP with the target like kidnapping, trapping, disable, etc, maybe it should need a rewording on the objective? I feel like people read the "prevent X from escaping *alive* on the shuttle* and think it's asassinate 2.0

**Comment by DatFrostys** [comment]
Same as above, mixed feelings.
I like the idea of getting out of your comfort zone to RP with a protection target but I imagine these problems may rise :
The protection targets is willing to RP properly
The protection target is SSD
The protection target killed themselves
The protection target fucked off to another Z level
The protection target is a head of department/security department player.

Again, having this forced interaction between traitor and target + other traitors in what may be epic fights is cool but I can't help but feel that those problems above will make the experience more frustrating than fun.

**Comment by CydiaLamiales** [comment]
Please PLEASE if you're doing this, unfuck the check for the completion of the 'protect' objective, it fails 90% of the time.

**Comment by Absolucy** [comment]
> Please PLEASE if you're doing this, unfuck the check for the completion of the 'protect' objective, it fails 90% of the time.

@CydiaLamiales In the code, I don't see any potentials issue with the protect objective, unless `considered_alive` is fucked up. Any more context to this?

@Fronsis:

> i feel like this will lead to a lot of scenarios where traitors will decide to act "pro-nt" just to protect their objective

They still might have an objective to like steal the nuke disk or kill some sec officer or something. Traitors do get _2_ normal objectives + gimmick + escape/hijack/DAGD.

> i can already predict a player going to sec, claim their allegiance but that they've been tasked to protect X and promise to not cause issues

Again, this falls apart, if they care about their protect objective, they're most likely going to care about their other objective.

Perhaps I could make it so if the first objective is protect, the second objective is guaranteed to _not_ be protect, to avoid situations where someone has 100% "benevolent" objectives.

> there should be a reason why the Syndicate randomly wants X person to be protected when they have no allegiance with the Syndicate

The Syndicate is canonically loose and disorganized as hell. For all we know, someone outside the Syndicate could be paying someone to protect them.

@DatFrostys:

>  The protection target is SSD

I could make it so the target being SSD will succeed the protect objective.

>  The protection target killed themselves

Protect already does succeed if the target uses the Suicide verb.

>  The protection target fucked off to another Z level

Emergent gameplay, you can't always win.  
Also, `/datum/objective/proc/find_target` is actually biased _against_ picking miners or explorers as target, for this reason.

> The protection target is a head of department/security department player.

Perfect roleplay opportunity, try to get yourself assigned as their bodyguard or something, I've seen such a thing succeed multiple times.

**Comment by CydiaLamiales** [comment]
> @CydiaLamiales In the code, I don't see any potentials issue with the protect objective, unless considered_alive is fucked up. Any more context to this?

I don't know exactly where the issue here is, but I don't think I've ever seen a protect objective succeed
image
image

**Comment by llol111** [comment]
While i entirely agree that kill objectives should be shaken up somewhat, i do not like protect objectives.
Entirely subjective personal oppinion that, of course. But they just aint verry fun in my experience. 

My personaly prefered solution reworking kill objectives would be to have like a 40% Chance or whatever to replace kill objectives with less severe "Teach X a lesson" objectives, where on the mechanical layer you just need to beat them up some or kill them (with revives being fine)

**Comment by Absolucy** [comment]
> My personaly prefered solution reworking kill objectives would be to have like a 40% Chance or whatever to replace kill objectives with less severe "Teach X a lesson" objectives, where on the mechanical layer you just need to beat them up some or kill them (with revives being fine)

@llol111 I do remember some other server having a kill objective variant which was basically "Ensure [person] has died at least once."

**Comment by Fronsis** [comment]
> > My personaly prefered solution reworking kill objectives would be to have like a 40% Chance or whatever to replace kill objectives with less severe "Teach X a lesson" objectives, where on the mechanical layer you just need to beat them up some or kill them (with revives being fine)
> 
> @llol111 I do remember some other server having a kill objective variant which was basically "Ensure [person] has died at least once."
> 
> Could be a good idea, as it would discourage (although not _prevent_) full-on round removal.

I believe SR has that sort of objectives with certain antags whenever they don't choose the ''create your own objective'' prompt, i feel like it's dependant of roleplay, like ''Sorry you're gonna die to pay of your debt to the Syndicate'' and then if you met him again you could RP that the deeds is done and they shouldn't mess with the Syndicate again

This would certainly improve to prevent the round-removals, there's too many people that been abusing the bs bag lately..

Though i remember i proposed that idea and a lot of people were like ''but why would they make you kill them once/twice!?'' saying it didn't make sense

**Comment by itsmeow** [comment]
> Please PLEASE if you're doing this, unfuck the check for the completion of the 'protect' objective, it fails 90% of the time.

I fixed this in #8487

**Comment by Isy232** [comment]
I'd argue 55% chance for protect objective to be targeted is still too low.
I'd suggest 75%.

If theres an "almost half" chance your target is not targetted at all, I feel that would make for a high chance of rather easy and boring objectives.1

File: code/modules/antagonists/traitor/datum_traitor.dm [hunk 1]
```diff
@@ -155,16 +160,10 @@
 			destroy_objective.owner = owner
 			destroy_objective.find_target()
 			add_objective(destroy_objective)
-		else if(prob(30))
-			var/datum/objective/maroon/maroon_objective = new
-			maroon_objective.owner = owner
-			maroon_objective.find_target()
-			add_objective(maroon_objective)
+		else if (prob(50))
```
Comment (itsmeow) [inline code/modules/antagonists/traitor/datum_traitor.dm]: Should definitely be like 30 or so. Protect should not be a primary objective. Maroon/kill is much better.
```suggestion
		else if (prob(30))
```

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
TMing to see how it plays

**Comment by PowerfulBacon** [comment]
Protect is a frustrating objective since you have to babysit your target the entire round compared to other objectives which allow for preparation, execution and cleanup. Protect objectives are just execution and nothing else. Although it could be interesting to have traitor conflict, I think this will be more frustrating than anything

**Comment by itsmeow** [comment]
I think it's decent enough but there are better routes to go when it comes to objectives. I agree that Protect does not exactly have good gameplay, although I *do* like the traitor vs traitor component which could potentially be adapted into something else.

**Pull request closed without merging**